src/dswx_sar/metadata.py

```diff
@@ -0,0 +1,311 @@
+import os
+import h5py
+import glob
+from datetime import datetime
+from collections import defaultdict
+
+import numpy as np
+from dswx_sar.version import VERSION as SOFTWARE_VERSION
+from dswx_sar.dswx_sar_util import read_geotiff, band_assign_value_dict
+
+# Constants
+UNKNOWN = 'UNKNOWN'
+PRODUCT_VERSION = 'UNKNOWN'
+DEFAULT_METADATA = {
+    'DSWX_PRODUCT_VERSION': PRODUCT_VERSION,
+    'SOFTWARE_VERSION': SOFTWARE_VERSION,
+    'PROJECT': 'OPERA',
+    'PRODUCT_LEVEL': '3',
+    'PRODUCT_TYPE': UNKNOWN,
+    'PRODUCT_SOURCE': UNKNOWN,
+    'SPACECRAFT_NAME': UNKNOWN,
+    'SENSOR': UNKNOWN
+}
+
+
+def _copy_meta_data_from_rtc(h5path_list, dswx_metadata_dict):
+    """Copy metadata dictionary from RTC metadata.
+    
+    Parameters
+    ----------
+    h5path_list : list
+        List of metadata HDF5 file paths.
+    dswx_metadata_dict : collections.OrderedDict
+        Metadata dictionary to populate.
+    """
+
+    metadata_dict = defaultdict(list)
+
+    metadata_rtc_fields = {
+        'orbitPassDirection': 'str',
+        'burstID': 'str',
+        'productVersion': 'str',
+        'zeroDopplerStartTime': 'str',
+        'trackNumber': 'int',
+        'absoluteOrbitNumber': 'int'
+    }
+
+    # Mapping from RTC metadata to DSWx metadata
+    dswx_meta_mapping = {
+        'orbitPassDirection': 'ORBIT_PASS_DIRECTION',
+        'burstID': 'BURST_ID',
+        'productVersion': 'RTC_PRODUCT_VERSION',
+        'zeroDopplerStartTime': 'SENSING_TIME',
+        'trackNumber': 'TRACK_NUMBER',
+        'absoluteOrbitNumber': 'ABSOLUTE_ORBIT_NUMBER'
+    }
+
+    for h5_meta_path in h5path_list:
+        with h5py.File(h5_meta_path) as src:
+            for field, dtype in metadata_rtc_fields.items():
+                if dtype == 'str':
+                    value = str(src[f'/identification/{field}'].asstr()[...])
+
+                else:
+                    value = int(src[f'/identification/{field}'][...])
```

Comment:
Any chance that the metadata value is in floating point?

Reply:
Thank you for your concern. But I don't think it can happen. 
We are reading the following information from RTC. So, the `trackNumber` and `absoluteOrbitNumber` should be integer. 
```
    metadata_rtc_fields = {
        'orbitPassDirection': 'str',
        'burstID': 'str',
        'productVersion': 'str',
        'zeroDopplerStartTime': 'str',
        'trackNumber': 'int',
        'absoluteOrbitNumber': 'int'
    }
```